stan/math/prim/scal/prob/normal_sufficient_lpdf.hpp

```diff
@@ -0,0 +1,175 @@
+#ifndef STAN_MATH_PRIM_SCAL_PROB_NORMAL_SUFFICIENT_LPDF_HPP
+#define STAN_MATH_PRIM_SCAL_PROB_NORMAL_SUFFICIENT_LPDF_HPP
+
+#include <stan/math/prim/scal/meta/return_type.hpp>
+#include <stan/math/prim/scal/prob/normal_lpdf.hpp>
+
+#include <stan/math/prim/scal/meta/OperandsAndPartials.hpp>
+#include <stan/math/prim/scal/meta/scalar_seq_view.hpp>
+#include <stan/math/prim/scal/err/check_consistent_sizes.hpp>
+#include <stan/math/prim/scal/err/check_finite.hpp>
+#include <stan/math/prim/scal/err/check_positive.hpp>
+#include <stan/math/prim/scal/err/check_nonnegative.hpp>
+#include <stan/math/prim/scal/fun/constants.hpp>
+#include <stan/math/prim/scal/fun/value_of.hpp>
+#include <stan/math/prim/scal/meta/include_summand.hpp>
+#include <stan/math/prim/scal/meta/VectorBuilder.hpp>
+#include <stan/math/prim/scal/meta/max_size.hpp>
+
+namespace stan {
+
+  namespace math {
+
+    /**
+     * The log of the normal density for the specified scalar(s) given
+     * the specified mean(s) and deviation(s).
+     * y, s_quared, mu, or sigma can each be either
+     * a scalar, a std vector or Eigen vector.
+     * n can be either a single int or an std vector of ints.
+     * Any vector inputs must be the same length.
+     *
+     * <p>The result log probability is defined to be the sum of the
+     * log probabilities for each observation/mean/deviation triple.
+     * @param y_bar (Sequence of) scalar(s) (sample average(s)).
+     * @param s_squared (Sequence of) sum(s) of sample squared errors
+     * @param n_obs (Sequence of) sample size(s)
+     * @param mu (Sequence of) location parameter(s)
+     * for the normal distribution.
+     * @param sigma (Sequence of) scale parameters for the normal
+     * distribution.
+     * @return The log of the product of the densities.
+     * @throw std::domain_error
+     * if either n or sigma are not positive,
+     * if s_squared is negative or
+     * if any parameter is not finite.
+     * @tparam T_y Underlying type of scalar in sequence.
+     * @tparam T_loc Type of location parameter.
+     */
+    template <bool propto,
+              typename T_y, typename T_s, typename T_n, typename T_loc,
+              typename T_scale>
+    typename return_type<T_y, T_s, T_loc, T_scale>::type
+    normal_sufficient_lpdf(const T_y& y_bar, const T_s& s_squared,
+                          const T_n& n_obs, const T_loc& mu,
```

Comment:
* `const` on second line should line up with first line.

* the `tparam` come first, not last

* space before the param doc

* minimize the number of lines---for example, the four lines of `throw std::domain_error` should fit on two lines